forge-game/src/main/java/forge/game/combat/AttackConstraints.java

```diff
@@ -405,7 +412,11 @@ public final int countViolations(final Map<Card, GameEntity> attackers) {
         }
 
         int violations = 0;
+
         for (final Card possibleAttacker : possibleAttackers) {
+            if (possibleAttacker.getController().mustAttack() && attackers.size() < 1) {
+                violations++;
```

Comment:
the AI doesn't seem to care if it has a violation or two – I think I am misunderstanding what this code does

I also tried `return -1;` here which puts Forge into a forever loop of `AI Attack declaration invalid, defaulting to: {}` when the AI doesn't want to attack